### PR TITLE
Fix 5243 Skip repair replica not in colocate group.

### LINKED ISSUE #5243: [BUG]Endless colocate table tablet repair
**Describe the bug**
A colocate tablet replica group is [A B C], the tablet's replicas are in [A B C D]
If D was removed from cluster and the replica on D is bad, the tablet's state will be VERSION_INCOMPLETE.
Doris will never fix this tablet.

**To Reproduce**
1. Create a colocate table
2. Remove a BE where replicas are placed
3. Load some data(before deleting the replica in the BE)

**Expected behavior**
1. The tablet's state became to `COLOCATE_MISMATCH`
2. Copy some replica to other BE.
3. [*BUG*]The tablet's state became to `VERSION_INCOMPLETE`, but we can not fix it forever.

**Screenshots**
image
image

**Desktop:**
 - Version 0.12(not fix in master)

### PR DESCRIPTION
**Pull request by liutang123**
## Proposed changes
Fix #5243  Skip repair replica not in colocate group.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

- [x] I have created an issue on (Fix #5243) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [] I have added tests that prove my fix is effective or that my feature works
- [] If this change need a document change, I have updated the document
- [] Any dependent changes have been merged

### CODE REVIEW

**Comment by kangkaisen** [comment]
@liutang123 Hi, Please add a UT for getColocateHealthStatus. BTW. The `visibleVersionHash` ars could be removed.

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Tablet.java [hunk 1]
```diff
@@ -547,14 +547,16 @@ public TabletStatus getColocateHealthStatus(long visibleVersion, long visibleVer
...
+        if (!replicaBackendIds.containsAll(backendsSet)) {
+            return TabletStatus.COLOCATE_MISMATCH;
         }
 
         // 2. check version completeness
         for (Replica replica : replicas) {
+            if (!backendsSet.contains(replica.getBackendId())) {
+                // We don't care about replicas that are not in backendsSet
+                continue;
+            }
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Tablet.java:559]: Why not fall down to COLOCATE_REDUNDANT?

Comment (morningman) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/Tablet.java:559]: Oh I miss understood

**Review by morningman — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Tablet.java [hunk 2]
```diff
@@ -542,19 +542,20 @@ public long getDataSize(boolean singleReplica) {
...
-                return TabletStatus.COLOCATE_MISMATCH;
-            }
+        if (!replicaBackendIds.containsAll(backendsSet)) {
+            return TabletStatus.COLOCATE_MISMATCH;
         }
 
         // 2. check version completeness
         for (Replica replica : replicas) {
+            if (!backendsSet.contains(replica.getBackendId())) {
+                // We don't care about replicas that are not in backendsSet
```
Comment (wangbo) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Tablet.java:556]:
```suggestion
                // We don't care about replicas that are not in backendsSet.
                // eg:  replicaBackendIds=(1,2,3,4); backendsSet=(1,2,3), then replica 4 should be skipped here and then goto ```COLOCATE_REDUNDANT``` in step 3
```

Comment (morningman) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/Tablet.java:556]: Hi @liutang123 , modify it?

**Review by wangbo — APPROVED** [review]
+1

**Pull request merged**